File: skid_steer_ws/src/skid_steer_py_examples/skid_steer_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__("Simple_publisher")
        self.pub = self.create_publisher(String, "Chatter", 10)
        self.counter = 0
        self.frequency = 1.0
        self.get_logger().info("Publishing at %d Hz" % self.frequency)
        self.timer = self.create_timer(self.frequency, self.TimerCallback)
    def TimerCallback(self):
        msg = String()
        msg.data = "Hello ROS2 - counter: %d" % self.counter
        self.pub.publish(msg)
        self.counter += 1

def main():
    rclpy.init()
    simple_publisher = SimplePublisher()
    rclpy.spin(simple_publisher)
    simple_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == '__name__':
    main()

File: skid_steer_ws/src/skid_steer_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>
using namespace std::chrono_literals;
class SimplePublisher : public rclcpp::Node{
public:
    SimplePublisher() : Node("Simple_Publisher"), counter{0}
    {
        pub = create_publisher<std_msgs::msg::String>("Chatter", 10);
        timer = create_wall_timer(1s, std::bind(&SimplePublisher::TimerCallback, this));
        RCLCPP_INFO(get_logger(), "Publishing at 1 Hz");
    }

private:
    unsigned int counter;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub;
    rclcpp::TimerBase::SharedPtr timer;

    void TimerCallback(){
        auto message = std_msgs::msg::String();
        message.data = "Hello ROS2 - counter: " + std::to_string(counter++);
        pub->publish(message);
    }
};


int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}